### PR TITLE
Add support for automated validation

### PR DESCRIPTION
**Pull request by ptitzler**
This PR
 - adds a Makefile with lint targets and its dependencies
 - adds a Git workflow that invokes lint
 - updates `.gitignore`

### CODE REVIEW

**Comment by ptitzler** [comment]
Any suggestions for/experience with notebook linters?

**Comment by ptitzler** [comment]
Note that `GitHub Actions has encountered an internal error when running your job.`
[hopefully] (temporary) GitHub issue: `https://github.com/actions/virtual-environments/issues/2179#issuecomment-841268363`

Also note that even after ^^ is fixed validation will fail because the existing `.py` file isn't valid. This PR does not make an attempt to address that issue.

**Comment by ptitzler** [comment]
Tested with https://github.com/nbQA-dev/nbQA, which seems to work. The nice things is that it's basically a wrapper that can invoke `flake8` and therefore applies the same rules.

File: .github/workflows/build.yaml [hunk 1]
```diff
@@ -0,0 +1,49 @@
+#
+# Copyright 2021-2021 Elyra Authors
```
Comment (lresende) [inline .github/workflows/build.yaml:2]:
```suggestion
# Copyright 2018-2021 Elyra Authors
```

File: Makefile [hunk 2]
```diff
@@ -0,0 +1,34 @@
+#
+# Copyright 2021-2021 Elyra Authors
```
Comment (lresende) [inline Makefile:2]:
```suggestion
# Copyright 2018-2021 Elyra Authors
```

File: .github/workflows/build.yaml (hunk 1 above)
Comment (ptitzler) [reply .github/workflows/build.yaml:2]: I followed this rule, assuming that the scope of the copyright is this file, not Elyra in general:

> The 'start' year should be the date of the oldest content still contained in the work, with the 'end' year being the last year that the work was revised.

File: Makefile (hunk 2 above)
Comment (ptitzler) [reply Makefile:2]: See above

**Comment by ptitzler** [comment]
Changes incorporated @lresende

**Review by lresende — APPROVED** [review]

**Pull request merged**